core/chains/evm/headtracker/head_saver.go

```diff
@@ -14,15 +14,15 @@ type headSaver struct {
 	orm    ORM
 	config Config
 	logger logger.Logger
-	heads  Heads
+	heads  heads[*evmtypes.Head, common.Hash]
 }
 
 func NewHeadSaver(lggr logger.Logger, orm ORM, config Config) httypes.HeadSaver {
 	return &headSaver{
 		orm:    orm,
 		config: config,
 		logger: lggr.Named("HeadSaver"),
-		heads:  NewHeads(),
+		heads:  *NewEvmHeads(),
```

Comment:
why use dereference here?

couldn't the `NewEvmHeads` and `NewHeads` function just return the struct rather than taking a reference and then immediately dereferencing?

Reply:
thanks for pointing that out. My mistake for taking the reference and referencing it again, returning a struct is a much better method.